superset-frontend/src/explore/components/DataTablesPane/index.tsx

```diff
@@ -298,7 +299,7 @@ export const DataTablesPane = ({
       ...prevState,
       [RESULT_TYPES.samples]: true,
     }));
-  }, [queryFormData?.adhoc_filters, queryFormData?.datasource]);
+  }, [queryFormData?.adhoc_filters, queryFormData?.datasource, panelOpen]);
```

Comment:
Why do we make it dependent on `panelOpen`? I don't think we should rerun fetching samples on opening and closing the panel.
Also, I think that it shouldn't depend on `queryFormData?.adhoc_filters`.

Reply:
because the source table schema might be changed. the sample table design by `get sample from source table` so we should do through query every time.

Reply:
I see, but refetching on every panel open feels very suboptimal. Can we use `state.explore.datasource` from Redux instead? I think it holds the currently selected dataset, regardless of the latest query

Reply:
The key problem to solve here is that when the schema of table changed, we don't have a chance to refresh the sample cache

Reply:
I remember when I was implementing the data tables panel that minimizing the number of requests was a big deal for many people and I'm afraid that it will be considered a regression when we send a new request on each opening of the panel

Reply:
Thanks for the reminder! I think we need to make a tradeoff. BTW, there is a setting item: `SAMPLES_ROW_LIMIT = 1000` in config.py that how many sample rows will return.